Userland/Utilities/du.cpp

```diff
@@ -182,7 +192,7 @@ u64 print_space_usage(DeprecatedString const& path, DuOption const& du_option, s
 
         while (di.has_next()) {
             auto const child_path = di.next_full_path();
-            size += print_space_usage(child_path, du_option, current_depth + 1);
+            size += print_space_usage(child_path, du_option, current_depth + 1, root_device);
```

Comment:
:thinking: shouldn't it be enough to just check if the child has a different `st_dev` here?

```cpp
if (!du_option.one_file_system || (path_stat.st_dev == child_path.st_dev)) {
    size += ...
}
```